### PR TITLE
Bitbucket OAuth enablement

### PR DESCRIPTION
**Pull request by Magisus**
PR to fix bugs found while testing OAuth tokens with bitbucket.

### CODE REVIEW

**Comment by justinstoller** [comment]
What's up with this?

**Comment by Magisus** [comment]
I could go either way on it. It's fine to go in, but I think there are other issues with Bitbucket OAuth besides this. Let me update the commit to reference the ticket at least.

**Comment by Magisus** [comment]
I had one more commit to fix a general bug related to auth retries. Once that is fixed, the underlying issue with Bitbucket is that it expects a different username along with the token. I think how we fix that could be a bigger question, to avoid special-casing in this code.

File: lib/r10k/git/rugged/credentials.rb [hunk 1]
```diff
@@ -67,9 +67,14 @@ def get_plaintext_credentials(url, username_from_url)
     end
 
     if token_path = per_repo_oauth_token || R10K::Git.settings[:oauth_token]
-      if token_path == '-'
+      if @oauth_token
+        token = @oauth_token
```
Comment (justinstoller) [inline lib/r10k/git/rugged/credentials.rb]: Is it just me or is setting a token to the value of an instance variable in one branch and in the next branch setting the same instance variable to the value of the same token confusing?

IMO, it'd help readability if we moved this conditional and the token validation into a separate function (say `foo_fn`) and conditionally called it like
```rb
@oauth_token ||= foo_fn()
```
... and then set `password` to `@oauth_token`.

Comment (Magisus) [reply lib/r10k/git/rugged/credentials.rb]: It's definitely terrible, I never really meant to put this up (and didn't look at it when I pushed it just now), I just wanted to get something working. I'll see about refactoring it.

Comment (Magisus) [reply lib/r10k/git/rugged/credentials.rb]: Updated, with a test, lemme know what you think.

**Review by justinstoller — APPROVED** [review]

**Review by mwaggett — APPROVED** [review]

**Pull request merged**